From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>
Bug: https://github.com/haskell/haskell-language-server/pull/2212

ghcide: Add flags to toggle building each executable

This package is a transient dependency of `haskell-language-server`.
However, only the library is needed. These three flags give the option for
those who do not need the executables to disable them. All three are
enabled by default.

`executable`:
* Toggles the `ghcide` executable

`test-exe`:
* Toggles the `ghcide-test-preprocessor` executable

`bench-exe`:
* Toggles the `ghcide-bench` executable

diff -urN ghcide-1.4.0.3.orig/ghcide.cabal ghcide-1.4.0.3/ghcide.cabal
--- ghcide-1.4.0.3.orig/ghcide.cabal	2021-07-13 12:14:55.926090165 -0600
+++ ghcide-1.4.0.3/ghcide.cabal	2021-07-13 12:18:02.918221483 -0600
@@ -203,7 +203,13 @@
     if flag(ghc-patched-unboxed-bytecode)
       cpp-options: -DGHC_PATCHED_UNBOXED_BYTECODE
 
+flag test-exe
+    description: Build the ghcide-test-preprocessor executable
+    default: True
+
 executable ghcide-test-preprocessor
+    if !flag(test-exe)
+        buildable: False
     default-language: Haskell2010
     hs-source-dirs: test/preprocessor
     ghc-options: -Wall -Wno-name-shadowing
@@ -249,7 +255,13 @@
         text,
         yaml
 
+flag executable
+    description: Build the ghcide executable
+    default: True
+
 executable ghcide
+    if !flag(executable)
+        buildable: False
     default-language:   Haskell2010
     hs-source-dirs:     exe
     ghc-options:
@@ -390,7 +402,13 @@
         TypeApplications
         ViewPatterns
 
+flag bench-exe
+    description: Build the ghcide-bench executable
+    default: True
+
 executable ghcide-bench
+    if !flag(bench-exe)
+        buildable: False
     default-language: Haskell2010
     build-tool-depends:
         ghcide:ghcide
